flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -231,7 +247,13 @@ static constexpr IntrinsicHandler handlers[]{
     {"ceiling", &I::genCeiling},
     {"char", &I::genChar},
     {"conjg", &I::genConjg},
-    {"date_and_time", &I::genDateAndTime},
+    {"date_and_time",
+     &I::genDateAndTime,
+     {{{"date", asAddr},
+       {"time", asAddr},
+       {"zone", asAddr},
+       {"values", asAddr}}},
+     /*isElemental*/ false},
```

Comment:
We probably need to start thinking about tablegen soon.

Reply:
That would be a good candidate indeed.